examples/window-app/nrfconnect/main/WindowCovering.cpp

```diff
@@ -52,24 +63,6 @@ WindowCovering::WindowCovering()
     {
         LOG_ERR("Cannot initialize the tilt indicator");
     }
-
-    k_timer_init(&sLiftTimer, MoveTimerTimeoutCallback, nullptr);
-    k_timer_init(&sTiltTimer, MoveTimerTimeoutCallback, nullptr);
-}
-
-void WindowCovering::MoveTimerTimeoutCallback(k_timer * aTimer)
-{
-    if (!aTimer)
-        return;
-
-    if (aTimer == &sLiftTimer)
-    {
-        chip::DeviceLayer::PlatformMgr().ScheduleWork(DriveCurrentLiftPosition);
-    }
-    else if (aTimer == &sTiltTimer)
-    {
-        chip::DeviceLayer::PlatformMgr().ScheduleWork(DriveCurrentTiltPosition);
-    }
```

Comment:
What was the issue exactly with this timer ?
I prefer this implementation which factorize rather than your proposal 
What was the issue motivating this PR ?

Reply:
A timer was called in an interrupt, and during it, a mutex was used. Unfortunately, Zephyr doesn't allow to get mutex within an interrupt, so it caused an assertion and, as a result, a hard fault.
`ASSERTION FAIL [!arch_is_in_isr()] @ WEST_TOPDIR/zephyr/kernel/mutex.c:101
mutexes cannot be used inside ISRs
`
The fix is to move a timer invocation to the `SystemLayer` context, and then a problem with mutex will disappear. To do this I should call `SystemLayer().StartTimer` function and then a `MoveTimerTimeoutCallbackLift/Tilt` can be used.

What do you mean: "I prefer this implementation which factorize"? I had to change a timer's callback implementation because we can't use it in this context anymore.

Reply:
Got it ! Then can you pass argument for Tilt/Lift within the pointer here set to nullprt ?  MoveTimerTimeoutCallbackLift, nullptr);  ,i guess it goes to void * appState ? I don't know those API but seems to be possible